algorithms/scaling/algorithm.py

```diff
@@ -264,11 +266,9 @@ def remove_bad_data(self):
         if removed_ids:
             logger.info("deleting removed datasets from memory: %s", removed_ids)
             expids = list(self.experiments.identifiers())
-            locs_in_list = []
-            for id_ in removed_ids:
-                locs_in_list.append(expids.index(id_))
+            locs_in_list = [expids.index(expid) for expid in expids]
             self.experiments, self.reflections = select_datasets_on_ids(
-                self.experiments, self.reflections, exclude_datasets=removed_ids
+                self.experiments, self.reflections, exclude_datasets=locs_in_list
```

Comment:
`removed_ids` no longer appears to be used? Is this definitely correct? Is this code not already tested?

Reply:
Ugh, sorry, that's a typo.  Thanks for spotting.

This code doesn't seem to be adequately tested.  It fails for non-integer experiment IDs, such as the new-style UID strings.

Reply:
I can imagine this was a genuine bug introduced by the UID changes, I would say this now looks correct.